RecursiveExtractor/Extractors/TarExtractor.cs

```diff
@@ -63,7 +65,12 @@ public async IAsyncEnumerable<FileEntry> ExtractAsync(FileEntry fileEntry, Extra
                     {
                         Logger.Debug(Extractor.FAILED_PARSING_ERROR_MESSAGE_STRING, ArchiveFileType.TAR, fileEntry.FullPath, tarEntry.Key, e.GetType());
                     }
-                    var name = tarEntry.Key.Replace('/', Path.DirectorySeparatorChar);
+                    var name = tarEntry.Key?.Replace('/', Path.DirectorySeparatorChar);
+                    if (string.IsNullOrEmpty(name))
+                    {
+                        Logger.Debug(Extractor.ENTRY_MISSING_NAME_ERROR_MESSAGE_STRING, ArchiveFileType.TAR, fileEntry.FullPath);
+                        continue;
+                    }
```

Comment:
Unrelated to this PR's changes but if opening the tarEntry's stream or copying the stream to the file fails / throws an exception, can we reason about the state of the backing stream?

It doesn't seem strictly right to go ahead and use it within the `FileEntry` constructor call (which performs some stream operations) and you're likely to end up with duplicate errors/logs from different places in the code for the same stream error.

Reply:
That's a good point. It probably makes sense to bail out earlier after the exception on opening/copying the entry rather than continue trying to parse it. Agree that it didn't really fit in this PR, but I may have time in the next week or two to revisit this and the other improvements you recommended. https://github.com/microsoft/RecursiveExtractor/issues/156